pkg/health/checker/checker.go

```diff
@@ -41,28 +68,45 @@ type healthStore interface {
 	GetServiceHealth(service string) (map[string]consul.WatchResult, error)
 }
 
-type healthKV interface {
-	List(prefix string, opts *api.QueryOptions) (api.KVPairs, *api.QueryMeta, error)
+type ReplicationControllerStore interface {
+	Get(id rcfields.ID) (rcfields.RC, error)
 }
 
-type consulHealthChecker struct {
-	client      consulutil.ConsulClient
-	kv          healthKV
-	consulStore healthStore
+type LabelReader interface {
+	GetMatches(klabels.Selector, labels.Type) ([]labels.Labeled, error)
 }
-type consulHealth interface {
-	Node(types.NodeName, *api.QueryOptions) ([]*api.HealthCheck, *api.QueryMeta, error)
+
+type healthChecker struct {
+	healthClient   HealthClient
+	resourceClient ResourceClient
+	consulClient   consulutil.ConsulClient
+	kv             healthKV
+	consulStore    healthStore
+	rcStore        ReplicationControllerStore
+	labelReader    LabelReader
+}
+
+func NewShadowTrafficHealthChecker(hClient HealthClient, resourceClient ResourceClient, cClient consulutil.ConsulClient, rcStore ReplicationControllerStore, labelReader LabelReader) HealthChecker {
+	return healthChecker{
+		healthClient:   hClient,
+		resourceClient: resourceClient,
+		consulClient:   cClient,
+		kv:             cClient.KV(),
+		consulStore:    consul.NewConsulStore(cClient),
+		rcStore:        rcStore,
+		labelReader:    labelReader,
+	}
 }
 
-func NewConsulHealthChecker(client consulutil.ConsulClient) ConsulHealthChecker {
-	return consulHealthChecker{
-		client:      client,
-		kv:          client.KV(),
-		consulStore: consul.NewConsulStore(client),
+func NewHealthChecker(cClient consulutil.ConsulClient) HealthChecker {
+	return healthChecker{
```

Comment:
it still makes me nervous that these two constructors return structs of the same type. If I instantiate a `healthChecker` by calling `NewHealthChecker()`, and then I pass `true` for `useHealthService`, i'm going to end up with a nil pointer exception at runtime. If you use different types to implement the functionality, the "normal" health checker could just ignore that argument

Reply:
Thanks for the reminder, I was going to get around to that.